### ISSUE TITLE
jump to definition doesn't work if code deployed in conda environment other than 'base'

### ISSUE THREAD

**Issue by stellarshenson** [jupyterlab/jupyterlab#16680]
## Description

When using `jump to definition` feature, while code was installed in the environment other than `base`, message `No jump targets found` is displayed.  Autocompletion works fine.

When the same code is installed in `base` environment, `jump to definition` works fine. 

## Reproduce

1. Have a project folder with `src` folder and modules in it and `pyproject.toml` to provide build instructions and dependencies
2. Install this project using `pip install --editable .`
3. Install `nb_conda_kernels` extension
4. Create new environment with conda, call it `new_env`
5. Create new notebook, import your module from `src`
6. Select `new_env` kernel for the notebook
7. Call function from your module, test autocompletion (should work fine)
8. Attempt to jump to funciton definition, observe `No jump targets found`

## Expected behavior

Module python file should be opened with the function definition to jump to.
This works only if the module was installed in `base` environment. At the same time autocomplete works fine.

## Context

- Operating System and version: Linux Ubuntu 22.04 
- Browser and version: Chrome
- JupyterLab version: 4.2.4

</details>

**Comment by krassowski** [comment]
> jump to definition

Do you mean the one from [`jupyterlab-lsp`](https://github.com/jupyter-lsp/jupyterlab-lsp) extension? Or another extension?

**Comment by stellarshenson** [comment]
Yes this oneKonrad JeleńOn 14 Aug 2024, at 17:34, Michał Krassowski ***@***.***> wrote:﻿

jump to definition

Do you mean the one from jupyterlab-lsp extension? Or another extension?

**Comment by JasonWeill** [comment]
@stellarshenson Thanks for opening this issue! Per conversation with @krassowski , let's continue the conversation in the JupyterLab LSP extension's repo, where there are already a few bugs concerning the jump to definition feature.

**Comment by JasonWeill** [comment]
@stellarshenson I can't move your issue directly to https://github.com/jupyter-lsp/jupyterlab-lsp/issues/ . Can you please open a new issue in this other project, providing all info requested there? Thanks again!
